daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -191,7 +197,7 @@ class ContractStateMachine[Nid](mode: ContractKeyUniquenessMode) {
             // If the key was brought in scope before, we must update `keys`
             // independently of whether this exercise is by-key because it affects later key lookups.
             if (mode.byKeyOnly) {
-              keys.get(gkey).orElse(globalKeyInputs.get(gkey).map(_.toKeyMapping)) match {
+              keys.get(gkey).orElse(lookupActiveGlobalKeyInput(gkey)) match {
```

Comment:
I think this one actually doesn’t matter. The contract cannot have been consumed twice so we’ll never run into a case where the activeness check makes a difference. But checking "everything goes through lookupActiveGlobalKeyInput" is easier than trying to be clever.